pkg/devfile/parser/configurables.go

```diff
@@ -0,0 +1,253 @@
+		Name:   NameDescription,
+		Ports:  PortsDescription,
+		Memory: MemoryDescription,
+	}
+
+	lowerCaseDevfileParameters = util.GetLowerCaseParameters(GetDevfileSupportedParameters())
+)
+
+// SetConfiguration allows setting all the parameters that are configurable in a devfile
+func (d DevfileObj) SetConfiguration(parameter string, value interface{}) error {
+
+	// we are ignoring this error becase a developer is usually aware of the type of value that is
+	// being passed. So consider this a shortcut, if you know its a string value use this strValue
+	// else parse it inside the switch case.
+	strValue, _ := value.(string)
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.setMetadataName(strValue)
+		case "ports":
+			arrValue := strings.Split(strValue, ",")
+			return d.setPorts(arrValue...)
+		case "memory":
+			return d.setMemory(strValue)
+		}
+
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a configurable parameter in the devfile", parameter)
+
+}
+
+// DeleteConfiguration allows deleting  the parameters that are configurable in a devfile
+func (d DevfileObj) DeleteConfiguration(parameter string) error {
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.setMetadataName("")
+		case "ports":
+			return d.removePorts()
+		case "memory":
+			return d.setMemory("")
+		}
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a configurable parameter in the devfile", parameter)
+}
+
+// IsSet checks if a parameter is set in the devfile
+func (d DevfileObj) IsSet(parameter string) bool {
+
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.getMetadataName() != ""
+		case "ports":
+			return d.hasPorts()
+		case "memory":
+			return d.getMemory() != ""
+		}
+	}
+	return false
+
+}
+
+func (d DevfileObj) setMetadataName(name string) error {
+	metadata := d.Data.GetMetadata()
+	d.Data.SetMetadata(name, metadata.Version)
+	return d.WriteYamlDevfile()
+}
+
+// AddEnvVars adds environment variables to all the components in a devfile
+func (d DevfileObj) AddEnvVars(otherList config.EnvVarList) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			currentlist := config.NewEnvVarListFromDevfileEnv(component.Container.Env)
+			component.Container.Env = currentlist.Merge(otherList).ToDevfileEnv()
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+// RemoveEnvVars removes the environment variables which have the keys from all the components in a devfile
+func (d DevfileObj) RemoveEnvVars(keys []string) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+
+			currentlist := config.NewEnvVarListFromDevfileEnv(component.Container.Env)
```

Comment:
I'm obviously very late on this but making the devfile implementation depend on the s2i types is a very bad idea…